/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = 'proto3';

package spark.connect;

import "spark/connect/expressions.proto";

option java_multiple_files = true;
option java_package = "org.apache.spark.connect.proto";

// MlParams stores param settings for
// ML Estimator / Transformer / Model / Evaluator
message MlParams {
  // user-supplied params
  map<string, Expression.Literal> params = 1;
}

// represents a reference to server side `Model` instance
message ModelRef {
  // The ID is used to lookup the model instance in server side.
  string id = 1;
}

// Message for fetching model attribute. It can be representing to a Relation or a ML command
// Eg, model.coefficients, model.summary.weightedPrecision
// or model.summary.roc which returns a DataFrame
message FetchModelAttr {
  // (Required) reference to the server model object
  ModelRef model_ref = 1;
  // (Required) the method name. Eg, "coefficients" or "summary.weightedPrecision"
  string method = 2;
}